apps/portal/modules/user.js

```diff
@@ -19,9 +19,18 @@ var roles;
     login = function (username, password) {
         var carbon = require('carbon');
         var server = new carbon.server.Server();
-        if (!server.authenticate(username, password)) {
-            return false;
+        try {
+            if (!server.authenticate(username, password)) {
+                return false;
+            }
+        }catch (e){
+            var message = "Invalid Domain Name";
```

Comment:
There might be other reasons the exception could have been thrown, and those cases are swallowed. Please log actual exception when and exception occurred.